### PR TITLE
Support tls openssl3

### PR DESCRIPTION
**Pull request by johnvenn**
upgrade support_tls branch to support openssl 3.x version based on support_openssl_3.0-dev

### CODE REVIEW

**Review by jinghe-INTC — APPROVED** [review]

File: Linux/package/include/sgx_tsgxssl.edl [hunk 1]
```diff
@@ -36,7 +36,10 @@ enclave {
     from "sgx_tstdc.edl" import *;
     
     untrusted {
-    	 void u_sgxssl_ftime([out, size=timeb_len] void * timeptr, uint32_t timeb_len);
-    };
+    	void u_sgxssl_ftime([out, size=timeb_len] void * timeptr, uint32_t timeb_len);
+		size_t u_sgxssl_write (int fd, [in,size=n]const void *buf, size_t n);
+	    size_t u_sgxssl_read(int fd, [out, size=count]void *buf, size_t count);
+        int u_sgxssl_close(int fd);
```
Comment (lzha101) [inline Linux/package/include/sgx_tsgxssl.edl:42]: The indent is not consistent.

File: Linux/sgx/libsgx_usgxssl/uunistd.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		return ret;
+	}
+	
+	size_t u_sgxssl_read(int fd, void* buf, size_t count)
+	{
+		size_t ret = read(fd, buf, count);
+		return ret;
+	}
+	
+	int u_sgxssl_close(int fd)
```
Comment (lzha101) [inline Linux/sgx/libsgx_usgxssl/uunistd.cpp:49]: Just curious why there is no ocall to open the file.  Is the fd maintained inside enclave or outside enclave?

Comment (johnvenn) [reply Linux/sgx/libsgx_usgxssl/uunistd.cpp:49]: this is one aspect that need to be enhanced in this support_tls branch. the branch is now only for SAmpleAttestedTLS, part of socket function is put in that sample, I will enhance the function to this branch for more general use.

File: Linux/package/include/sgx_tsgxssl.edl (hunk 1 above)
Comment (johnvenn) [reply Linux/package/include/sgx_tsgxssl.edl:42]: resolved, thanks for the review:)

File: Linux/sgx/libsgx_usgxssl/uunistd.cpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ */
+
+#include <unistd.h>
+#include "ucommon.h"
+
+extern "C" {
+
+	size_t u_sgxssl_write(int fd, const void* buf, size_t n)
+	{
+		size_t ret = write(fd, buf, n);
```
[the remark this answers is not in the document]
Comment (johnvenn) [reply Linux/sgx/libsgx_usgxssl/uunistd.cpp:39]: ok, I will update the branch

File: Linux/sgx/libsgx_tsgxssl/tunistd.cpp [hunk 4]
```diff
@@ -93,10 +99,16 @@ size_t sgxssl_read(int fd, void *buf, size_t count)
 	// In addition, the function is used by bss_sock.c as readsocket function.
 	// It is unreachable under the assumption that TLS support is not required.
 	// Otherwise should be implemented as OCALL.
-	SGX_UNREACHABLE_CODE(SET_ERRNO);
+	size_t retval = 0;
+	sgx_status_t ret = u_sgxssl_read(&retval, fd, buf, count);
+	if (ret != SGX_SUCCESS)
+	{
+		FEND;
+		return 0;
```
[the remark this answers is not in the document]
Comment (johnvenn) [reply Linux/sgx/libsgx_tsgxssl/tunistd.cpp:107]: you are right, I will submit another PR to address this issue.

**Pull request merged**